packages/adapter/addon/rest.js

```diff
@@ -309,18 +312,26 @@ const RESTAdapter = Adapter.extend(BuildURLMixin, {
     },
   }),
 
-  useFetch: computed(function() {
-    let ENV = getOwner(this).resolveRegistration('config:environment');
-    // TODO: https://github.com/emberjs/data/issues/6093
-    let jQueryIntegrationDisabled = ENV && ENV.EmberENV && ENV.EmberENV._JQUERY_INTEGRATION === false;
-
-    if (jQueryIntegrationDisabled) {
-      return true;
-    } else if (hasNajax || hasJQuery) {
-      return false;
-    } else {
-      return true;
-    }
+  useFetch: computed({
+    get() {
+      if (this._useFetch !== undefined) {
+        return this._useFetch;
+      }
+      let ENV = getOwner(this).resolveRegistration('config:environment');
+      // TODO: https://github.com/emberjs/data/issues/6093
+      let jQueryIntegrationDisabled = ENV && ENV.EmberENV && ENV.EmberENV._JQUERY_INTEGRATION === false;
+
+      if (jQueryIntegrationDisabled) {
+        return true;
+      } else if (hasNajax || hasJQuery) {
```

Comment:
we should move the hasNajax determination within something featureflagged, so something like

```js
else if (DEPRECATE_NAJAX && typeof najax !== 'undefined') {
  return false;
}
```

then move the jquery check into its own branch so we can strip that too on it's own later once possible.

You can see how this works here using: https://terser-playground.surge.sh/

```js
if (a) {
  c()
} else if (false && true) {
  d()
} else if (false && true) {
  e()
} else {
  b();
}
```